myapp/segmentsconfigapp/parser.py

```diff
@@ -0,0 +1,78 @@
+import re
+
+class ParserSegmentConfig():
+
+    """
+    Lexer
+    """
+
+    def token_validation(self, mystr):
+        list_pretoken = mystr.split('\n')
+        for pretoken in list_pretoken:
+            if re.match("^@{1}[a-zA-Z_0-9]+$", pretoken):
+                self.token_li.append(Token("rulename", pretoken))
+            elif pretoken == '':
+                self.token_li.append(Token("separator", pretoken))
+            else:
+                self.token_li.append(Token("rule", pretoken))
+
+    """
+    Parser
+    """
+    
+    def parse_tokens(self):
+        self.token_iterli = iter(self.token_li)
+        try:
+            self.getrulename_list()
+        except SyntaxError as err:
+            self.last_error = err.args[0]
```

Comment:
use str(err) instead ?
```When an exception occurs, it may have an associated value, also known as the exception’s argument. The presence and type of the argument depend on the exception type.

The except clause may specify a variable after the exception name (or tuple). The variable is bound to an exception instance with the arguments stored in instance.args. For convenience, the exception instance defines __str__() so the arguments can be printed directly without having to reference .args.```